dev-games/trenchbroom/patches/trenchbroom-2025.1.patchset

```diff
@@ -0,0 +1,297 @@
+@@ -248,22 +248,24 @@ if(WIN32 OR ${CMAKE_SYSTEM_NAME} MATCHES "Linux|FreeBSD")
+             COMMAND ${CMAKE_COMMAND} -E make_directory "${APP_RESOURCE_DEST_DIR}/stylesheets/"
+             COMMAND ${CMAKE_COMMAND} -E copy_directory "${APP_RESOURCE_DIR}/stylesheets" "${APP_RESOURCE_DEST_DIR}/stylesheets")
+ 
+-    # Copy manual files to resource directory
+-    add_custom_command(TARGET TrenchBroom POST_BUILD
+-            COMMAND ${CMAKE_COMMAND} -E make_directory "${APP_RESOURCE_DEST_DIR}/manual/"
+-            COMMAND ${CMAKE_COMMAND} -E make_directory "${APP_RESOURCE_DEST_DIR}/manual/images"
+-            COMMAND ${CMAKE_COMMAND} -E copy "${INDEX_OUTPUT_PATH}" "${APP_RESOURCE_DEST_DIR}/manual/"
+-            COMMAND ${CMAKE_COMMAND} -E copy "${DOC_MANUAL_SHORTCUTS_JS_TARGET_ABSOLUTE}" "${APP_RESOURCE_DEST_DIR}/manual/")
+-
+-    foreach(MANUAL_FILE ${DOC_MANUAL_TARGET_FILES_ABSOLUTE})
++    if(NOT ${CMAKE_SYSTEM_NAME} MATCHES "Haiku")
++        # Copy manual files to resource directory
+         add_custom_command(TARGET TrenchBroom POST_BUILD
+-                COMMAND ${CMAKE_COMMAND} -E copy ${MANUAL_FILE} "${APP_RESOURCE_DEST_DIR}/manual/")
+-    endforeach(MANUAL_FILE)
++                COMMAND ${CMAKE_COMMAND} -E make_directory "${APP_RESOURCE_DEST_DIR}/manual/"
++                COMMAND ${CMAKE_COMMAND} -E make_directory "${APP_RESOURCE_DEST_DIR}/manual/images"
++                COMMAND ${CMAKE_COMMAND} -E copy "${INDEX_OUTPUT_PATH}" "${APP_RESOURCE_DEST_DIR}/manual/"
++                COMMAND ${CMAKE_COMMAND} -E copy "${DOC_MANUAL_SHORTCUTS_JS_TARGET_ABSOLUTE}" "${APP_RESOURCE_DEST_DIR}/manual/")
+ 
+-    foreach(MANUAL_IMAGE_FILE ${DOC_MANUAL_TARGET_IMAGE_FILES_ABSOLUTE})
+-        add_custom_command(TARGET TrenchBroom POST_BUILD
+-                COMMAND ${CMAKE_COMMAND} -E copy ${MANUAL_IMAGE_FILE} "${APP_RESOURCE_DEST_DIR}/manual/images/")
+-    endforeach(MANUAL_IMAGE_FILE)
++        foreach(MANUAL_FILE ${DOC_MANUAL_TARGET_FILES_ABSOLUTE})
++            add_custom_command(TARGET TrenchBroom POST_BUILD
++                    COMMAND ${CMAKE_COMMAND} -E copy ${MANUAL_FILE} "${APP_RESOURCE_DEST_DIR}/manual/")
++        endforeach(MANUAL_FILE)
++
++        foreach(MANUAL_IMAGE_FILE ${DOC_MANUAL_TARGET_IMAGE_FILES_ABSOLUTE})
++            add_custom_command(TARGET TrenchBroom POST_BUILD
++                    COMMAND ${CMAKE_COMMAND} -E copy ${MANUAL_IMAGE_FILE} "${APP_RESOURCE_DEST_DIR}/manual/images/")
++        endforeach(MANUAL_IMAGE_FILE)
++    endif()
+ endif()
+ 
+ # Common CPack configuration
+diff --git a/cmake/Utils.cmake b/cmake/Utils.cmake
+index 33233de95..7725b3549 100644
+--- a/cmake/Utils.cmake
++++ b/cmake/Utils.cmake
+@@ -228,13 +228,5 @@ macro(GET_GIT_DESCRIBE GIT SOURCE_DIR GIT_DESCRIBE)
+ endmacro(GET_GIT_DESCRIBE)
+ 
+ macro(GET_BUILD_PLATFORM PLATFORM_NAME)
+-    if(WIN32)
+-        set(${PLATFORM_NAME} "Win64")
+-    elseif(APPLE)
+-        set(${PLATFORM_NAME} "macOS")
+-    elseif(UNIX)
```

Comment:
You could use: `elseif(UNIX AND NOT HAIKU)` and add `a elseif(HAIKU)` after that instead of removing all the others.

Reply:
Checking further this patch is obsolete, Haiku is detected as UNIX, looking at the about box there is no change mentioned in regards of Haiku.

Reply:
I don't think that will work, UNIX will still override HAIKU because it takes Haiku as UNIX?

Reply:
If you put HAIKU before UNIX, the UNIX case will not be used. if-elseif cascades are evaluated in order and the first successful match is taken and nothing else (or the `else` if there is one and nothing matches).